Additions/UIView-KIFAdditions.m

```diff
@@ -247,32 +253,28 @@ - (UIAccessibilityElement *)accessibilityElementMatchingBlock:(BOOL(^)(UIAccessi
                     NSIndexPath *indexPath = [NSIndexPath indexPathForRow:row inSection:section];
                     if ([indexPathsForVisibleRows containsObject:indexPath]) {
                         continue;
-                    }
-                    
-                    @autoreleasepool {
-                        // Get the cell directly from the dataSource because UITableView will only vend visible cells
-                        UITableViewCell *cell = [tableView.dataSource tableView:tableView cellForRowAtIndexPath:indexPath];
-                        
-                        UIAccessibilityElement *element = [cell accessibilityElementMatchingBlock:matchBlock notHidden:NO];
-                        
-                        // Remove the cell from the table view so that it doesn't stick around
-                        [cell removeFromSuperview];
-                        
-                        // Skip this cell if it isn't the one we're looking for
-                        if (!element) {
-                            continue;
-                        }
-                    }
-                    
-                    // Scroll to the cell and wait for the animation to complete
-                    [tableView scrollToRowAtIndexPath:indexPath atScrollPosition:UITableViewScrollPositionNone animated:YES];
-                    // Note: using KIFRunLoopRunInModeRelativeToAnimationSpeed here may cause tests to stall
-                    CFRunLoopRunInMode(UIApplicationCurrentRunMode, 0.5, false);
-                    
+					}
+
+					@autoreleasepool {
+						// Scroll to the cell and wait for the animation to complete
+						[tableView scrollToRowAtIndexPath:indexPath atScrollPosition:UITableViewScrollPositionNone animated:NO];
```

Comment:
The intent of the existing logic is to not scroll the tableview unless it is known that an element within that tableview should be made visible. 

Otherwise, we're taking a potentially huge performance penalty to scroll through all elements in table views to try and find an element that happens to be in the view hierarchy after the table view. This will still happen even if the table view is completely offscreen, right?

Have you noticed this when running your full suite of tests? Maybe it isn't so bad due to not animating the tableview scrolling.

Reply:
Performance hit is minimal.